src/eva/utilities/utils.py

```diff
@@ -474,3 +475,93 @@ def is_number(s):
         return False
 
 # --------------------------------------------------------------------------------------------------
+
+
+def generate_filenames_from_template(config, logger=None):
+
+    """
+    Generate a list of filenames using either curly-brace or strftime-style datetime formatting.
+
+    This function constructs a list of file paths by interpolating datetime strings
+    into a filename template using one of two styles:
+    - Curly-brace format: e.g., 'file_{datetime}.nc'
+    - strftime-style format: e.g., 'file.%Y%m%d%H.nc'
+
+    Parameters:
+        config (dict): A dictionary containing:
+            - template (str): A strftime-style template string, e.g., "/data/gdas.%Y%m%d/%H/..."
+            - start (str): Start time string in "%Y%m%d%H" format.
+            - end (str): End time string in "%Y%m%d%H" format.
+            - interval_hours (int): Interval (in hours) between timestamps.
+
+        logger (Logger, optional): The logger object for logging messages. Defaults to None.
+
+    Returns:
+        list: A list of fully formatted filenames.
+
+    Example:
+        Case 1:
+        {
+            "template": "/data/file_%Y%m%d%H.nc",
+            "start": "2024010100",
+            "end": "2024010112",
+            "interval_hours": 6,
+        }
+
+        ➜ ["/data/file_2024010100.nc", "/data/file_2024010106.nc", "/data/file_2024010112.nc"]
+
+        Case 2:
+        {
+            "template": "gdas.%Y%m%d/%H/atmos/gdas.t%Hz.atmanl.nc",
+            "start": "2024010100",
+            "end": "2024010112",
+            "interval_hours": 6,
+            "datetime_format": "%Y%m%d%H"
+        }
+
+        ➜ ["gdas.20240101/00/atmos/gdas.t00z.atmanl.nc", ...]
+    """
+    def assert_datetime_format(date_str, fmt="%Y%m%d%H", label="date"):
```

Comment:
If using the ISO format in the YAML:
datetime.fromisoformat(date_string) should do the trick and simplify things?